expression/builtin_other.go

```diff
@@ -187,12 +324,50 @@ func (b *builtinInStringSig) evalInt(row chunk.Row) (int64, bool, error) {
 
 // builtinInRealSig see https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_in
 type builtinInRealSig struct {
-	baseBuiltinFunc
+	baseInSig
+	hashSet map[float64]bool
+}
+
+func (b *builtinInRealSig) buildHashMapForConstArgs(ctx sessionctx.Context) error {
+	b.nonConstArgs = make([]Expression, 0, len(b.args))
+	b.nonConstArgs = append(b.nonConstArgs, b.args[0])
+	b.hashSet = make(map[float64]bool, len(b.args)-1)
+	count := 0
+	for i := 1; i < len(b.args); i++ {
+		if b.args[i].ConstItem(b.ctx.GetSessionVars().StmtCtx) {
+			val, isNull, err := b.args[i].EvalReal(ctx, chunk.Row{})
+			if err != nil {
+				return err
+			}
+			if isNull {
+				b.hasNull = true
+				continue
+			}
+			b.hashSet[val] = true
```

Comment:
ditto, we can use `Float64Set` defined in `util/set/float64_set.go`, compared with `map[float64]bool`, it consumes less memory.